java/runtime/src/main/java/io/ray/runtime/gcs/GcsClient.java

```diff
@@ -147,6 +147,10 @@ public boolean wasCurrentActorRestarted(ActorId actorId) {
     return actorTableData.getNumRestarts() != 0;
   }
 
+  public byte[] getActorAddress(ActorId actorId) {
+    return globalStateAccessor.getActorAddress(actorId);
+  }
```

Comment:
I wanted to use the existing getActorInfo method and extract the actor rpc address. But I couldn't find how to use it in JNI. Any suggestions?

Reply:
Maybe we can avoid this?  some stuff like, pass the owner actor or actor id in `nativePut()`.

Reply:
Maybe I misunderstood your words. Do you want to get the address in `nativePut()`? 

The pseudo is:
```diff
   /**
diff --git a/src/ray/core_worker/lib/java/io_ray_runtime_object_NativeObjectStore.cc b/src/ray/core_worker/lib/java/io_ray_runtime_object_NativeObjectStore.
cc
index f14ced4f7..4d260494b 100644
--- a/src/ray/core_worker/lib/java/io_ray_runtime_object_NativeObjectStore.cc
+++ b/src/ray/core_worker/lib/java/io_ray_runtime_object_NativeObjectStore.cc
@@ -86,9 +86,24 @@ Java_io_ray_runtime_object_NativeObjectStore_nativePut__Lio_ray_runtime_object_N

 JNIEXPORT jbyteArray JNICALL
 Java_io_ray_runtime_object_NativeObjectStore_nativePut__Lio_ray_runtime_object_NativeRayObject_2_3B(
-    JNIEnv *env, jclass, jobject obj, jbyteArray java_actor_address) {
+    JNIEnv *env, jclass, jobject obj, jbyteArray owner_actor_id_bytes) {
   ray::ObjectID object_id;
-  auto address = JavaByteArrayToNativeString(env, java_actor_address);
+
+  ray::rpc::ActorTableData actor_table_data;
+  {
+    /// Get actor info from GCS synchronously.
+    std::vector<std::string> actor_table_data_bytes;
+    std::promise<bool> promise;
+    auto gcs_client = ray::CoreWorkerProcess::GetCoreWorker().GetGcsClient();
+    RAY_CHECK_OK(gcs_client->Actors().AsyncGetAll(
+        ActorID::FromBinary(JavaBytesToNativeString(owner_actor_id_bytes));
+        TransformForMultiItemCallback<rpc::ActorTableData>(actor_table_data_bytes,
+                                                           promise)));
+    promise.get_future().get();
+    actor_table_data.ParseFromString(actor_table_data_bytes);
+  }
+
+  auto address = actor_table_data.address();
   auto status = PutSerializedObject(env, obj, /*object_id=*/ray::ObjectID::Nil(),
                                     /*out_object_id=*/&object_id, /*pin_object=*/true,
                                     /*owner_address=*/&address);
```

Reply:
Thanks, this way is much better.

Reply:
`TransformForMultiItemCallback` is declared as private, so I have to paste the call back here.